videozen-frontend/src/reducer/data-reducer.js

```diff
@@ -0,0 +1,144 @@
+import uuid from "react-uuid";
+
+const findPlaylistName = (allPlaylists, videoId) => {
+  // return allPlaylists.map((playlist) =>
+  //   playlist.videos.filter((video, playlist) =>
+  //     video.id === videoId ? playlist.name : ""
+  //   )
+  // );
+
+  for (let i = 0; i < allPlaylists.length; i++) {
+    for (let j = 0; j < allPlaylists[i].videos.length; j++) {
+      if (allPlaylists[i].videos[j].id === videoId) {
+        return allPlaylists[i].name;
+      }
+    }
+  }
+};
+
+export const dataReducer = (state, action) => {
+  const playlistName = findPlaylistName(
+    state.allPlaylists,
+    action.payload.video.id
+  );
+
+  switch (action.type) {
+    case "CREATE_NEW_PLAYLIST":
+      return {
+        ...state,
+        allPlaylists: state.allPlaylists.map((playlist) =>
+          playlist.name === playlistName
+            ? {
+                ...playlist,
+                videos: playlist.videos.map((video) =>
+                  video.id === action.payload.video.id
+                    ? {
+                        ...video,
+                        partOfPlaylists: [
+                          ...video.partOfPlaylists,
+                          action.payload.name,
+                        ],
+                      }
+                    : video
```

Comment:
Can move all this functionalities into seperate functions 
It makes code much more readable